witwidget/notebook/colab/wit.py

```diff
@@ -115,11 +117,27 @@ def compute_custom_distance(wit_id, index, callback_name, params):
       window.backendError = error => {{
         wit.handleError(error.msg);
       }};
-      window.inferenceCallback = inferences => {{
-        wit.labelVocab = inferences.label_vocab;
-        wit.inferences = inferences.inferences;
+      window.inferenceStartCallback = (inferences) => {{
+        stagedInferences = inferences;
+      }};
+      window.inferenceCallback = res => {{
+        for (let i = 0; i < res.results.length; i++) {{
+          if (wit.modelType == 'classification') {{
+            stagedInferences.inferences.results[i].classificationResult.classifications =
+              stagedInferences.inferences.results[i].classificationResult.classifications.concat(res.results[i]);
+          }}
+          else {{
+            stagedInferences.inferences.results[i].regressionResult.regressions =
+              stagedInferences.inferences.results[i].regressionResult.regressions.concat(res.results[i]);
+          }}
+        }}
+        stagedInferences.inferences.indices = stagedInferences.inferences.indices.concat(res.indices);
+      }};
+      window.inferenceEndCallback = () => {{
+        wit.labelVocab = stagedInferences.label_vocab;
+        wit.inferences = stagedInferences.inferences;
         wit.extraOutputs = {{indices: wit.inferences.indices,
-                             extra: inferences.extra_outputs}};
+                             extra: stagedInferences.extra_outputs}};
```

Comment:
Do extra outputs also get passed correctly? Do we chunk attributions as well or send them in a single chunk?

Reply:
great point. it is all passed in the first ("start") message now still, which causes crashes if large examples have extra output (which includes attributions). let me fix that.